src/playwright/test.ts

```diff
@@ -0,0 +1,348 @@
+import { promises as fs } from 'node:fs';
+import {
+  type Locator,
+  test as baseTest,
+  expect as baseExpect,
+} from '@playwright/test';
+import { type TemplateResult } from 'lit';
+import { render } from '@lit-labs/ssr';
+import { collectResult } from '@lit-labs/ssr/lib/render-result.js';
+import { AxeBuilder } from '@axe-core/playwright';
+import customElements from '../../custom-elements.json' with { type: 'json' };
+import { v8CoverageAttachmentName } from './constants.js';
+
+// Anything the comes from the browser will get serialized before it gets sent
+// back to tests, which run in Node.js. So `addEventListener()` can only return
+// a subset of event properties. But these should cover most of our use cases.
+interface SimpleEvent {
+  bubbles: boolean;
+  cancelable: boolean;
+  composed: boolean;
+  defaultPrevented: boolean;
+  detail?: unknown;
+  target: string | null;
+}
+
+export const expect = baseExpect.extend({
+  async toBeFocused(locator: Locator) {
```

Comment:
Also added a few [custom matchers](https://playwright.dev/docs/test-assertions#add-custom-matchers-using-expectextend) for things we commonly do.